### PR TITLE
refactor(sync): header sync mode

### PR DESCRIPTION
**Pull request by rkrasiuk**
Remove forkchoice from `Consensus` trait and simplify headers sync mode semantics. 
Some of these changes are taken from #1662, but constant conflicts became a nightmare to resolve.

NOTE: This PR detaches the channel between Engine API and the forkchoice/tip receiver. This will be fixed in the #1662.

### CODE REVIEW

**Review by onbjerg — COMMENTED** [review]
How would this work with other consensus algorithms? The reason a channel works for beacon is because consensus lives outside of EL, for PoW this is another story. For AuRa and Clique too I would imagine

**Comment by rkrasiuk** [comment]
@onbjerg that part i didn't change, just refactored, because `continuous` stage never polled the channel. agreed that we should find a better solution, but comment is irrelevant to this PR

**Comment by gakonst** [comment]
> How would this work with other consensus algorithms? The reason a channel works for beacon is because consensus lives outside of EL, for PoW this is another story. For AuRa and Clique too I would imagine

At this point I think we bite the bullet and go with it imo.

File: crates/stages/src/sets.rs [hunk 1]
```diff
@@ -130,6 +131,9 @@ where
 /// themselves offline.
 #[derive(Debug)]
 pub struct OnlineStages<H, B> {
+    /// The tip receiver for the first stage.
+    /// The download mode for the f
```
Comment (gakonst) [inline crates/stages/src/sets.rs:135]: Is this off?

File: crates/stages/src/stages/headers.rs [hunk 2]
```diff
@@ -112,28 +114,27 @@ where
...
+            HeaderSyncMode::Tip(ref mut rx) => {
+                loop {
+                    let _ = rx.changed().await; // TODO: remove this await?
+                    let tip = rx.borrow();
+                    if !tip.is_zero() {
+                        return SyncTarget::Tip(*tip)
+                    }
+                }
+            }
+            HeaderSyncMode::Continuous => {
```
Comment (gakonst) [inline crates/stages/src/stages/headers.rs:135]: Good!

**Review by gakonst — APPROVED** [review]
Ultimately fine with me - open to breaking the abstraction a little bit as we're still figuring it out

**Pull request merged**